### PR TITLE
Archive goal

### PR DESCRIPTION
**Pull request by qubist**

### CODE REVIEW

File: ui-src/src/event-listeners/index.js [hunk 1]
```diff
@@ -14,17 +14,31 @@ import {
...
+      case 'Escape':
         store.dispatch(closeGoalCreator())
         store.dispatch(unselectAll())
         break
+      case 'Backspace':
+        // archives one goal for now FIXME: should be able to archive many goals
+        let ui = store.getState().ui
+        // only dispatch if something's selected the createGoal window is
+        // not open
+        if (ui.selection.selectedGoals.length > 0 && !ui.goalCreation.isOpen) {
```
Comment (Connoropolous) [inline ui-src/src/event-listeners/index.js]: nice checks

**Comment by Connoropolous** [comment]
You'll need to change 
https://github.com/h-be/acorn/pull/12/files#diff-58b4899f563003a969caf55988a2c9a6R20
and
https://github.com/h-be/acorn/pull/12/files#diff-58b4899f563003a969caf55988a2c9a6R28

to be `payload.entry` instead of `payload.goal`

**Comment by qubist** [comment]
@Connoropolous this one's ready to be looked at

File: dna-src/zomes/my_zome/code/src/lib.rs [hunk 2]
```diff
@@ -239,7 +238,30 @@ mod my_zome {
...
                 LinkMatch::Any,
             )?
+            .into_iter().map(|edge: Edge| {
+                // re-create the goal entry to find its address
+                let address = Entry::App(
+                    "edge".into(),
+                    edge.clone().into(),
+                ).address();
+                // return a response structs with the edge and its address
+                GetResponse{entry: edge, address} // TODO: rename goal in GetResponse to something else
```
Comment (Connoropolous) [inline dna-src/zomes/my_zome/code/src/lib.rs]:
```suggestion
                GetResponse{entry: edge, address}
```

File: ui-src/src/event-listeners/index.js [hunk 3]
```diff
@@ -14,17 +14,44 @@ import {
...
+      case 'KeyG':
         store.dispatch(setGKeyDown())
         break
-      case "Escape":
+      case 'Escape':
         store.dispatch(closeGoalCreator())
         store.dispatch(unselectAll())
         break
+      case 'Backspace':
+        // archives one goal for now FIXME: should be able to archive many goals
```
Comment (Connoropolous) [inline ui-src/src/event-listeners/index.js]: This next bit is a bit bulky to stay in the event listener file and in this function... maybe a TODO note to refactor that

**Pull request merged**